draft-ietf-quic-transport.md

```diff
@@ -1890,6 +1897,46 @@ different static key or server identifier.
 Note that Stateless Reset messages do not have any cryptographic protection.
 
 
+#### Connection IDs and Stateless Reset {#reset-cid}
+
+This design relies on the client sending a connection ID in all packets so that
+the server can use the connection ID from a packet to reset the connection.  A
+server that uses this design cannot allow clients to omit a connection ID.  That
+is, it cannot set the truncate_connection_id transport parameter
+({{transport-parameter-definitions}}) to 0.
+
+Ensuring that the server sends the correct connection ID in its stateless reset
+presents an additional challenge.  Truncation of the connection ID in packets
+that use the short header means that the entire connection ID might not be
+present in the packet that triggers the stateless reset.  Furthermore, the
+server cannot know how many octets from the connection ID that the client
+requested be included in the short header.
+
+A server is expected to have a minimum number of octets that are needed for
+correct routing.  Thus, the server can always know that at least this many
+octets of the packet it receives contain a connection ID.
+
+If the server uses a longer connection ID than the client, there is the
+potential for the additional octets to be used to reveal that the stateless
+reset is not a regular packet.  This is not a serious issue, as it is already
+possible to use other fields to distinguish them from other packet.
```

Comment:
Really?  I thought we'd been able to stick to that goal so far.

Reply:
I guess we have.  Assuming that we do packet number encryption.  Right now the packet number leaks quite a lot of information.

Reply:
The construction of a stateless reset requires that the server provide a packet number.  It is probably using a connection ID that it used in the past, so packets can be correlated with previous packets.  The difference in packet numbers is observable.  A server has to pick a random packet number, which likely means a large gap.  That might be used to distinguish a stateless reset.

Packet number encryption fixes that problem.  I think that this will change if we decide to do that.

Reply:
Fair enough. I don't grok your point about using a connection ID from the past. 

This is another way that a stateless reset might be detectable: while sending a stateless reset the server does not know what the client had negotiated for connection ID truncation in the server -> client direction. If the client required the server to send a connection ID, the stateless reset would likely not carry one.

Reply:
My point is that if there were packets sent on this connection in the past, then the packet numbers there and in the stateless reset might reveal something.

The point about truncation is addressed in this PR (3 paragraphs down).  It's ugly though.